art/preprocessing/expectation_over_transformation/tensorflow.py

```diff
@@ -0,0 +1,113 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2021
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module defines a base class for EoT in TensorFlow v2.
+"""
+from abc import abstractmethod
+import logging
+from typing import Optional, Tuple, TYPE_CHECKING
+
+from art.preprocessing.preprocessing import PreprocessorTensorFlowV2
+
+if TYPE_CHECKING:
+    import tensorflow as tf
+
+logger = logging.getLogger(__name__)
+
+
+class EoTTensorFlowV2(PreprocessorTensorFlowV2):
+    """
+    This module defines a base class for EoT in TensorFlow v2.
+    """
+
+    def __init__(
+        self, nb_samples: int, clip_values: Tuple[float, float], apply_fit: bool = False, apply_predict: bool = True,
+    ) -> None:
+        """
+        Create an instance of EoTTensorFlowV2.
+
+        :param nb_samples: Number of random samples per input sample.
+        :param clip_values: Tuple of float representing minimum and maximum values of input `(min, max)`.
+        :param apply_fit: True if applied during fitting/training.
+        :param apply_predict: True if applied during predicting.
+        """
+        super().__init__(is_fitted=True, apply_fit=apply_fit, apply_predict=apply_predict)
+
+        self.nb_samples = nb_samples
+        self.clip_values = clip_values
+        EoTTensorFlowV2._check_params(self)
+
+    @abstractmethod
+    def _transform(
+        self, x: "tf.Tensor", y: Optional["tf.Tensor"], **kwargs
+    ) -> Tuple["tf.Tensor", Optional["tf.Tensor"]]:
+        """
+        Internal method implementing the transformation per input sample.
+
+        :param x: Input samples.
+        :param y: Label of the samples `x`.
+        :return: Transformed samples and labels.
+        """
+        raise NotImplementedError
+
+    def forward(self, x: "tf.Tensor", y: Optional["tf.Tensor"] = None) -> Tuple["tf.Tensor", Optional["tf.Tensor"]]:
+        """
+        Apply corruptions to inputs `x`.
```

Comment:
I think this `forward` function is not only applied for corruptions but also for image_rotation and other transformations in the future. So could you please generalize this sentence.